src/train_qlora.py

```
@@ -159,67 +236,169 @@ def loss_fn(model):
+                    'step': step,
+                    'lora_rank': self.config.model.lora_rank,
+                    'lora_alpha': self.config.model.lora_alpha,
+                    'distrust_alpha': self.config.distrust.alpha,
+                }, f, indent=2)
+            
+            print(f"Checkpoint saved")
+            return
         
-        print(f"Saving checkpoint to {output_path}")
+        # Create checkpoint state
+        from src.checkpoints.checkpoint_state import Checkpoint
```

Ruff (0.14.6)

Redefinition of unused `Checkpoint` from line 29

(F811)